File: bank simulator/app/schemas/account.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.enums import AccountStatus, AccountSubType, ProductCategory


class AccountResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    account_number: str
    cif: str
    product_category: ProductCategory
    account_sub_type: AccountSubType
    product_code: str
    branch_code: str
    currency: str
    ledger_balance: Decimal
    hold_amount: Decimal
    overdraft_limit: Decimal
    available_balance: Decimal
    status: AccountStatus
    opened_at: datetime | None = None


class AccountListResponse(BaseModel):
    cif: str
    accounts: list[AccountResponse]
    total: int

File: bank-dashboard/src/types/account.ts
export type Account = {
  account_number: string;
  cif: string;
  product_category: string;
  currency: string;
  ledger_balance: string;
  available_balance: string;
  hold_amount: string;
  status: string;
};

export type AccountListResponse = {
  cif: string;
  accounts: Account[];
  total: number;
};

export type TransferInput = {
  from_account: string;
  to_account: string;
  amount: string;
  currency: string;
  reference: string;
  idempotency_key: string;
};
